### ISSUE TITLE
FP: manual_unwrap_or help code is not correct

### ISSUE THREAD

**Issue by ncaq** [rust-lang/rust-clippy#6960]
Lint name: manual_unwrap_or

I tried this code:

```rust
use std::panic;
fn main() {
    let default_hook = panic::take_hook();
    panic::set_hook(Box::new(move |panic_info| {
        let message = match panic_info.payload().downcast_ref::<&str>() {
            None => "message is None",
            Some(s) => s,
        };
        // clippy suggest below code, But it is compile error.
        // let message = panic_info
        //     .payload()
        //     .downcast_ref::<&str>()
        //     .unwrap_or("message is None");
        println!("{}", message);
        default_hook(panic_info);
    }));
}
```

I expected to see this happen: clippy don't warn or suggest correct code.

Instead, this happened: 

When don't fix clippy output.

```console
2021-03-24T15:53:11 ❯ cargo clippy
    Checking clippy-suggest-unwrap-or v0.1.0 (/home/ncaq/Downloads/clippy-suggest-unwrap-or)
warning: this pattern reimplements `Option::unwrap_or`
 --> src/main.rs:5:23
  |
5 |           let message = match panic_info.payload().downcast_ref::<&str>() {
  |  _______________________^
6 | |             None => "message is None",
7 | |             Some(s) => s,
8 | |         };
  | |_________^ help: replace with: `panic_info.payload().downcast_ref::<&str>().unwrap_or("message is None")`
  |
  = note: `#[warn(clippy::manual_unwrap_or)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_unwrap_or

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.15s
```

When replace to help code.

```console
2021-03-24T15:53:13 ❯ cargo clippy
    Checking clippy-suggest-unwrap-or v0.1.0 (/home/ncaq/Downloads/clippy-suggest-unwrap-or)
error[E0308]: mismatched types
  --> src/main.rs:13:24
   |
13 |             .unwrap_or("message is None");
   |                        ^^^^^^^^^^^^^^^^^ expected `&str`, found `str`
   |
   = note: expected reference `&&str`
              found reference `&'static str`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `clippy-suggest-unwrap-or`

To learn more, run the command again with --verbose.
```

### Meta

- `cargo clippy -V`: e.g. clippy 0.0.212 (e1884a8 2020-12-29)
- `rustc -Vv`:
  ```
  rustc 1.49.0 (e1884a8e3 2020-12-29)
  binary: rustc
  commit-hash: e1884a8e3c3e813aada8254edfa120e85bf5ffca
  commit-date: 2020-12-29
  host: x86_64-unknown-linux-gnu
  release: 1.49.0
  ```

**Comment by camsteffen** [comment]
I believe the fix is to check `expr_adjustments().is_empty()` on `s` in the match arm body.
